src/main/scala/bi/deep/SparkSchema.scala

```diff
@@ -2,21 +2,15 @@ package bi.deep
 
 import org.apache.spark.sql.types.{StructField, StructType}
 
-class SparkSchema(
-                   var dimensions: Array[StructField],
-                   var metrics: Array[StructField]) {
+case class SparkSchema(
+                        dimensions: Array[StructField],
+                        metrics: Array[StructField]) {
 
   lazy val mergedSchema: StructType = StructType(dimensions ++ metrics)
   lazy val structTypeDimensions: StructType = StructType(dimensions)
   lazy val structTypeMetrics: StructType = StructType(metrics)
 
-  def updateDimensions(newDimensions: Array[StructField]): Unit = {
-    dimensions = newDimensions
+  def updateDimensions(newDimensions: Array[StructField]): SparkSchema = {
+    SparkSchema(newDimensions, metrics)
```

Comment:
case classes in Scala have built in `copy` that do the same as your updateDimensions
```scala
schema.copy(dimensions=newDimensions)
```
is equivalent to
```scala
schema.updateDimensions(newDimensions)
```

so your method is a bit redundant